lib/ecto/adapters/connection.ex

```diff
@@ -12,6 +12,37 @@ defmodule Ecto.Adapters.Connection do
 
   use Behaviour
 
+  defmacro __using__(_) do
+    quote do
+
+      @behaviour Ecto.Adapters.Connection
+
+      def after_connect(conn, opts) do
```

Comment:
Let's please not inject this function. Define it in Ecto.Adapters.Connection and call it directly .